query-engine/core/src/query_graph_builder/extractors/query_arguments.rs

```diff
@@ -113,13 +125,29 @@ fn process_order_object(
                         _ => unreachable!(),
                     };
 
-                    Ok(Some(OrderBy::new(sf.clone(), path, sort_order)))
+                    Ok(Some(OrderBy::new(sf.clone(), path, sort_order, None)))
                 }
             }
         }
     }
 }
 
+fn sort_aggregation_from_order_object(object: ParsedInputMap) -> QueryGraphBuilderResult<(SortAggregation, SortOrder)> {
+    let (field_name, field_value) = object.into_iter().next().unwrap();
+    let value: PrismaValue = field_value.try_into()?;
+    let sort_order = match value.into_string().unwrap().to_lowercase().as_str() {
```

Comment:
This is now used in multiple places. Extractor for the SortOrder can go into a separate function.